pointcloud/src/main/scala/geotrellis/pointcloud/spark/triangulation/BoundaryDelaunay.scala

```diff
@@ -47,16 +47,87 @@ object BoundaryDelaunay {
 
     val triangles = new TriangleMap(halfEdgeTable)
 
+    def writeWKT(wktFile: String) = {
+      val indexToCoord = { i: Int => Point.jtsCoord2Point(dt.pointSet.getCoordinate(i)) }
+      val mp = geotrellis.vector.MultiPolygon(triangles.getTriangles.keys.map{ case (i,j,k) => Polygon(indexToCoord(i), indexToCoord(j), indexToCoord(k), indexToCoord(i)) })
+      val wktString = geotrellis.vector.io.wkt.WKT.write(mp)
+      new java.io.PrintWriter(wktFile) { write(wktString); close }
+    }
+
+    def navigateThis(e0: Int) = {
+      halfEdgeTable.navigate(e0, verts.apply(_), Map[Char, (String, Int => Int)](
+        'x' -> (("export current loop to loop.wkt", { einit =>
+          import halfEdgeTable._
+          var e = einit
+          val pts = collection.mutable.ListBuffer[Coordinate](verts(getSrc(e)))
+          do {
+            pts += verts(getDest(e))
+            e = getNext(e)
+          } while (e != einit)
+          val loop = Polygon(pts.map(Point.jtsCoord2Point(_)))
+          new java.io.PrintWriter("loop.wkt") { write(geotrellis.vector.io.wkt.WKT.write(loop)); close }
+          einit
+        })),
+        'c' -> (("print the center of the current triangle's circumscribing circle", { e =>
+          val predicates = new TriangulationPredicates(DelaunayPointSet(verts.toMap), halfEdgeTable)
+          import predicates._
+          import halfEdgeTable._
+          val (_, center, valid) = circleCenter(getDest(e), getDest(getNext(e)), getDest(getNext(getNext(e))))
+          println(s"Circle center: $center " + {if (valid) "[valid]" else "[invalid]"})
+          e
+        })),
+        't' -> (("export triangles to triangles.wkt", { e => 
+          val polys = collection.mutable.ListBuffer.empty[Polygon]
+          val trans = { i : Int => Point.jtsCoord2Point(verts(i)) }
+          triangles.triangleVertices.foreach { case (i, j, k) => {
+            polys += Polygon(trans(i), trans(j), trans(k), trans(i))
+          }}
+          val mp = MultiPolygon(polys)
+          new java.io.PrintWriter("triangles.wkt"){ write(geotrellis.vector.io.wkt.WKT.write(mp)); close }
+          e
+        }))
+      ))
+    }
+
     def circumcircleLeavesExtent(extent: Extent)(tri: HalfEdge): Boolean = {
       import dt.halfEdgeTable._
       import dt.predicates._
 
-      val center = circleCenter(getDest(tri), getDest(getNext(tri)), getDest(getNext(getNext(tri))))
-      val radius = center.distance(new Coordinate(dt.pointSet.getX(getDest(tri)), dt.pointSet.getY(getDest(tri))))
+      val (radius, center, valid) = circleCenter(getDest(tri), getDest(getNext(tri)), getDest(getNext(getNext(tri))))
       val ppd = new PointPairDistance
 
       DistanceToPoint.computeDistance(extent.toPolygon.jtsGeom, center, ppd)
-      ppd.getDistance < radius
+      !valid || ppd.getDistance < radius
```

Comment:
Should bail on the invalid case before doing unnecessary-at-that-point computation, something like

```scala
!valid || {
  val ppd = new PointPairDistance
  DistanceToPoint.computeDistance(extent.toPolygon.jtsGeom, center, ppd)
  ppd.getDistance < radius
}
```